bundler/spec/commands/lock_spec.rb

```diff
@@ -1392,23 +1387,13 @@
     build_repo4 do
       build_gem "unf_ext", "0.0.8.2"
 
-      build_gem "unf_ext", "0.0.8.2" do |s|
-        s.required_ruby_version = [">= 2.4", "< #{previous_ruby_minor}"]
-        s.platform = "x64-mingw32"
-      end
-
       build_gem "unf_ext", "0.0.8.2" do |s|
         s.required_ruby_version = [">= #{previous_ruby_minor}", "< #{current_ruby_minor}"]
         s.platform = "x64-mingw-ucrt"
       end
 
       build_gem "google-protobuf", "3.21.12"
 
-      build_gem "google-protobuf", "3.21.12" do |s|
-        s.required_ruby_version = [">= 2.5", "< #{previous_ruby_minor}"]
-        s.platform = "x64-mingw32"
-      end
-
```

Comment:
The above spec in particular becomes meaningless if we remove these specs, because the new situation without x64-mingw32 no longer has conflicting requirements.

I checked and the spec was introduced as part of https://github.com/rubygems/rubygems/pull/6365, for covering a fix in pub_grub.

However, these days this spec no longer covers that fix, because even if you revert the pub grub fix, the test still passes.

I bisected when this spec stopped covering that pub grub bug, and it was after b8e55087f026ffcc5ebf135ca6726daf0ec40b5d.

Anyways, I think it's fine to completely remove this spec now. The related fix was a pub grub issue unlikely to ever be reverted since it was just a "typo".

Reply:
That makes much more sense. :sweat_smile: :pray: